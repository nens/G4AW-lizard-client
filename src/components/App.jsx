import MainView from "./views/MainView";
export default MainView;
