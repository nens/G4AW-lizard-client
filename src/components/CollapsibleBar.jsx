import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactDOM from "react-dom";
import styles from "./styles/CollapsibleBar.css";

// A CollapsibleBar component.

class CollapsibleBar extends Component {
  constructor() {
    super();
    this.state = {};
  }
  componentDidMount() {}
  render() {
    const { colorCode, title, subTitle, icon, handleClick, open } = this.props;
    return (
      <div className={styles.CollapsibleBar} onClick={handleClick}>
        <i className={`${styles.Arrow} material-icons`}>
          {open ? "keyboard_arrow_down" : "keyboard_arrow_right"}
        </i>
        <span className={styles.Title}>{title}</span>
        {subTitle
          ? <span className={styles.Subtitle}>{subTitle}</span>
          : <span className={styles.ColorCode}>
              <svg width="20px" height="20px" viewBox="0 0 20 20">
                <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
                  <circle fill={colorCode} cx="10" cy="10" r="10" />
                </g>
              </svg>
            </span>}
      </div>
    );
  }
}

CollapsibleBar.propTypes = {
  colorCode: PropTypes.string,
  handleClick: PropTypes.func,
  open: PropTypes.bool,
  subTitle: PropTypes.string,
  title: PropTypes.string
};

export default CollapsibleBar;
