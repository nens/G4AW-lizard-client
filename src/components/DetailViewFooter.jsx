import PropTypes from "prop-types";
import React, { Component } from "react";
import { connect } from "react-redux";
import { translate, t } from "react-i18next";

// import styles from "./styles/Legend.css";
import styles from "./styles/DetailViewFooter.css";

import { selectPreviousParcel, selectNextParcel } from "../actions";

class DetailViewFooterComponent extends Component {
  render() {
    return (
      <div className={styles.Wrapper}>
        <PrevParcelButton handleClick={this.props.showPrev} />
        <NextParcelButton handleClick={this.props.showNext} />
      </div>
    );
  }
}

class PrevParcelButton extends Component {
  render() {
    const { handleClick } = this.props;
    return (
      <div onClick={handleClick} className={styles.PrevParcelButton}>
        <i className="material-icons">keyboard_arrow_left</i>
      </div>
    );
  }
}

class NextParcelButton extends Component {
  render() {
    const { handleClick } = this.props;
    return (
      <div onClick={handleClick} className={styles.NextParcelButton}>
        <i className="material-icons">keyboard_arrow_right</i>
      </div>
    );
  }
}

DetailViewFooterComponent.propTypes = {
  showNext: PropTypes.func,
  showPrev: PropTypes.func
};

function mapDispatchToProps(dispatch) {
  return {
    showPrev: () => selectPreviousParcel(dispatch),
    showNext: () => selectNextParcel(dispatch)
  };
}

const DetailViewFooter = connect(null, mapDispatchToProps)(
  DetailViewFooterComponent
);

export default translate()(DetailViewFooter);
