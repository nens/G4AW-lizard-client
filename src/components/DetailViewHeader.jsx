import { Map, TileLayer, WMSTileLayer } from "react-leaflet";
import PropTypes from "prop-types";
import React, { Component } from "react";
import ReactDOM from "react-dom";
import styles from "./styles/DetailViewHeader.css";

// A DetailViewHeader component.

class DetailViewHeader extends Component {
  constructor() {
    super();
    this.state = {};
  }
  componentDidMount() {}
  render() {
    const {
      title,
      halfMode,
      subTitle,
      handleBackButtonClick,
      latlonzoom,
      headerImage
    } = this.props;
    const { lat, lon, zoom } = latlonzoom ? latlonzoom : {};
    console.log("halfMode", halfMode);
    return (
      <div
        className={`${styles.DetailViewHeader} ${halfMode ? styles.TitleOnlyMode : styles.FullMode}`}
      >
        <div className={styles.Titles}>
          <p className={styles.Title}>{title}</p>
          {halfMode ? "" : <span className={styles.SubTitle}>{subTitle}</span>}
        </div>
        <div className={styles.ArrowBackIcon} onClick={handleBackButtonClick}>
          <i className="material-icons">arrow_back</i>
        </div>
        {headerImage
          ? <div
              style={{
                width: "100%",
                height: "100%",
                backdropFilter: "blur(5px)",
                backgroundSize: "cover",
                backgroundImage: `url(${headerImage})`
              }}
            />
          : ""}
        {latlonzoom
          ? <Map
              ref="mapElement"
              center={[lat, lon]}
              zoom={zoom}
              onMoveend={this._handlePanOrZoomEnd}
              zoomControl={false}
              style={{ height: "100%", zIndex: -1 }}
            >
              <TileLayer url="https://{s}.tiles.mapbox.com/v3/nelenschuurmans.iaa98k8k/{z}/{x}/{y}.png" />
            </Map>
          : ""}
      </div>
    );
  }
}

DetailViewHeader.propTypes = {
  fullMode: PropTypes.bool,
  handleBackButtonClick: PropTypes.func,
  headerImage: PropTypes.string,
  latlonzoom: PropTypes.object,
  subTitle: PropTypes.string,
  title: PropTypes.string
};

export default DetailViewHeader;
