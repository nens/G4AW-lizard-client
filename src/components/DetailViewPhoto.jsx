import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactDOM from "react-dom";
import styles from "./styles/DetailViewPhoto.css";
import MDSpinner from "react-md-spinner";

// A DetailViewPhoto component.

class DetailViewPhoto extends Component {
  constructor() {
    super();
    this.state = {
      width: window.innerWidth,
      height: window.innerHeight,
      imageStatus: "loading"
      // selected: this.props.defaultSelected ? true : false
    };
    this.updateDimensions = this.updateDimensions.bind(this);
    this.handleNextPhoto = this.handleNextPhoto.bind(this);
    this.handlePrevPhoto = this.handlePrevPhoto.bind(this);
    this.handleImageLoaded = this.handleImageLoaded.bind(this);
  }
  componentDidMount() {
    this.updateDimensions();
    window.addEventListener("resize", this.updateDimensions);
  }
  updateDimensions() {
    this.setState({
      width: window.innerWidth,
      height: window.innerHeight
    });
  }
  handleNextPhoto() {
    console.log("handleNextPhoto()");
  }
  handlePrevPhoto() {
    console.log("handlePrevPhoto()");
  }
  handleImageLoaded() {
    this.setState({
      imageStatus: "loaded"
    });
  }
  render() {
    const {
      currentPhoto,
      images,
      handleBackButtonClick,
      handlePrevPhoto,
      handleNextPhoto
    } = this.props;
    const { width, height, imageStatus } = this.state;
    const currentPhotoIdx = currentPhoto ? currentPhoto : 0;
    const photo = images[currentPhotoIdx];

    // Replace this with moment.js after my PR is merged.
    const a = new Date(photo.date);
    const months = [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec"
    ];
    const year = a.getFullYear();
    const month = months[a.getMonth()];
    const date = a.getDate();
    const hour = a.getHours();
    const min = a.getMinutes();
    const sec = a.getSeconds();
    const time = `${date} ${month} ${year} ${hour}:${min}`;

    return (
      <div>
        {currentPhotoIdx > 0
          ? <div className={styles.PrevPhoto} onClick={this.handlePrevPhoto}>
              <i className={`${styles.LeftArrowIcon} material-icons`}>
                keyboard_arrow_left
              </i>
            </div>
          : ""}
        {currentPhotoIdx < images.length - 1
          ? <div className={styles.NextPhoto} onClick={this.handleNextPhoto}>
              <i className={`${styles.RightArrowIcon} material-icons`}>
                keyboard_arrow_right
              </i>
            </div>
          : ""}

        <div
          style={{
            display: "flex",
            flexDirection: "column",
            height,
            width
          }}
        >
          <div className={styles.TopPanel}>
            <div onClick={handleBackButtonClick} className={styles.BackButton}>
              <i className="material-icons">arrow_back</i>
            </div>
            <div>{currentPhotoIdx + 1}/{images.length}</div>
          </div>
          <div className={styles.PhotoPanel}>
            <div className={styles.DetailViewPhoto}>
              <img
                src={photo.url}
                width={width}
                onLoad={this.handleImageLoaded}
              />
            </div>
          </div>
          <div className={styles.PhotoPropertiesPanel}>
            <div className={styles.PhotoProperties}>
              <p className={styles.General}>General</p>
              <table className={styles.PropertiesTable}>
                <tr>
                  <td className={styles.PropertiesTableFirstColumn}>
                    Date taken
                  </td>
                  <td>{time}</td>
                </tr>
              </table>
            </div>
          </div>
        </div>

        {imageStatus === "loading"
          ? <div>
              <MDSpinner
                singleColor="#03a9f4"
                style={{
                  position: "absolute",
                  top: "50%",
                  left: "50%"
                }}
              />
            </div>
          : ""}
      </div>
    );
  }
}

DetailViewPhoto.propTypes = {
  images: PropTypes.array,
  handleBackButtonClick: PropTypes.func
};

export default DetailViewPhoto;
