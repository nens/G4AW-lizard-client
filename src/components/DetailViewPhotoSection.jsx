import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactDOM from "react-dom";
import { translate } from "react-i18next";

import { DetailViewSection, DetailViewPhoto } from ".";

import { PHOTO_LIST } from "../../stories/helpers";

class DetailViewPhotoSection extends Component {
  render() {
    const { isOpen, handleClick, t, photo } = this.props;
    return (
      <DetailViewSection title={t("Most recent image")} isOpen={isOpen}>
        <DetailViewPhoto photo={photo} handleClick={handleClick} />
      </DetailViewSection>
    );
  }
}

export default translate()(DetailViewPhotoSection);
