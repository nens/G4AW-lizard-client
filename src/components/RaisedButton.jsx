import { translate } from "react-i18next";
import Ink from "react-ink";
import PropTypes from "prop-types";
import React, { PureComponent } from "react";
import ReactDOM from "react-dom";
import styles from "./styles/RaisedButton.css";

import {
  VelocityTransitionGroup,
  VelocityComponent,
  velocityHelpers
} from "velocity-react";

// A RaisedButton emphasizes important functions on in the app.

class RaisedButton extends PureComponent {
  constructor() {
    super();
    this.state = {};
  }
  componentDidMount() {}
  render() {
    const { t, disabled, buttonText, iconClass } = this.props;
    return (
      <div
        className={`${styles.RaisedButton} ${disabled ? styles.disabled : ""}`}
        onClick={disabled ? "" : this.props.handleOnClick}
      >
        {iconClass
          ? <i className={`material-icons ${styles.Icon}`}>{iconClass}</i>
          : ""}
        &nbsp;<span>{buttonText}</span>
        <Ink />
      </div>
    );
  }
}

RaisedButton.propTypes = {
  buttonText: PropTypes.string,
  disabled: PropTypes.bool,
  handleOnClick: PropTypes.func,
  iconClass: PropTypes.string
};

export { RaisedButton };
export default translate()(RaisedButton);
