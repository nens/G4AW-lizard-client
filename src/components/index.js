import CollapsibleBar from "./CollapsibleBar";
import DetailViewHeader from "./DetailViewHeader";
import DetailViewPhoto from "./DetailViewPhoto";
import DetailViewPhotoSection from "./DetailViewPhotoSection";
import DetailViewSection from "./DetailViewSection";
import DetailViewTable from "./DetailViewTable";
import DetailViewTableSection from "./DetailViewTableSection";
import DetailViewThumbnails from "./DetailViewThumbnails";
import DetailViewThumbnailsSection from "./DetailViewThumbnailsSection";
import FlatButton from "./FlatButton";
import FooterBar from "./FooterBar";
import GeolocateButtonBig from "./GeolocateButtonBig";
import GeolocateButtonSmall from "./GeolocateButtonSmall";
import HeaderBar from "./HeaderBar";
import InputField from "./InputField";
import LayerSelection from "./LayerSelection";
import Legend from "./Legend";
import LoginLogoutButton from "./LoginLogoutButton";
import LoginModalButtons from "./LoginModalButtons";
import LoginModalContent from "./LoginModalContent";
import LoginModal from "./LoginModal";
import MapComponent from "./MapComponent";
import Modal from "./Modal";
import PulsatingMarker from "./PulsatingMarker";
import RaisedButton from "./RaisedButton";
import SearchBar from "./SearchBar";
import SearchResultCardItem from "./SearchResultCardItem";
import SearchResultListItem from "./SearchResultListItem";
import SettingsViewGeneral from "./SettingsViewGeneral";
import SettingsViewHelp from "./SettingsViewHelp";
import SettingsViewUser from "./SettingsViewUser";
import SimpleLineChart from "./SimpleLineChart";
import SnackBar from "./SnackBar";
import TabBar from "./TabBar";
import TimeseriesChart from "./TimeseriesChart";
import ToggleSwitch from "./ToggleSwitch";
import ViewSwitchButton from "./ViewSwitchButton";

export {
  CollapsibleBar,
  DetailViewHeader,
  DetailViewPhoto,
  DetailViewPhotoSection,
  DetailViewSection,
  DetailViewTable,
  DetailViewTableSection,
  DetailViewThumbnails,
  DetailViewThumbnailsSection,
  FlatButton,
  FooterBar,
  GeolocateButtonBig,
  GeolocateButtonSmall,
  HeaderBar,
  InputField,
  LayerSelection,
  Legend,
  LoginLogoutButton,
  LoginModalButtons,
  LoginModalContent,
  LoginModal,
  MapComponent,
  Modal,
  PulsatingMarker,
  RaisedButton,
  SearchBar,
  SearchResultCardItem,
  SearchResultListItem,
  SettingsViewGeneral,
  SettingsViewUser,
  SettingsViewHelp,
  SimpleLineChart,
  SnackBar,
  TabBar,
  TimeseriesChart,
  ToggleSwitch,
  ViewSwitchButton
};
