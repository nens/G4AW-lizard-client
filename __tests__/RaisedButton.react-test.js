// RaisedButton.react-test.js
import React from "react";
import { RaisedButton } from "../src/components/RaisedButton";
import renderer from "react-test-renderer";

test("buttonText is shown as OK", () => {
  const component = renderer.create(<RaisedButton buttonText="OK" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
