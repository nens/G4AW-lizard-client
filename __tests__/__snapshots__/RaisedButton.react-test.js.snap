// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonText is shown as OK 1`] = `
<div
  className="RaisedButton "
  onClick={undefined}
>
  
  Â 
  <span>
    OK
  </span>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</div>
`;
